interview-questions.md

```diff
@@ -8,33 +8,33 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Object-oriented programming is centered around objects rather than logic. This feature makes Ruby more user-friendly.
 
-Researched answer:
+Researched answer: Object-oriented programming is a particular programming paradigm centered and organized around objects. Ruby is an object-oriented programming language know for its simplicity and easy to read style. This helps programmers to section off pieces of code that perform particular tasks so the code can interact with many small parts. 
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: Ruby contains two types of numbers. Integers which are whole numbers such as 10 and floats which are numbers that include decimal points such as 10.5. It would be more beneficial to use floats when dealing with percentages tha include decimal places and integers when you want to round numbers to the nearest whole number.
 
-Researched answer:
+Researched answer: The difference bewtween a float and an integer in Ruby is integers are whole numbers meaning they have no decimals while floats contain decimals 
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: implicit return means that in the abscence of a return command the last line coded will be returned automatically. 
 
-Researched answer:
+Researched answer: implicit return means that in Ruby a return command is not necessary to return a line of code. The terminal will automatically return the last line of the method. For example if I had a method called "message" and in the next line I simply typed "You are the best", I will get the following string in return : "You are the best". 
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: A block works like a method in ruby but it does not belong to an object.
 
-Researched answer:
+Researched answer: A block funtions similarly as a method but does not belong to an object. Blocks can take in arguments and return a particular value and contains chunks of code. Blocks are usually enclosed in a do-end statment and take in arguments that are defined in between two pipe lines.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: Ruby hashes contains keys and the corresponding value for each key. To get the value the developer must type in ".values" to return only the values
 
-Researched answer:
+Researched answer: Ruby hashes are collections of unique key value pairs. The hash is similar to an arry but its values are not reference by indexes. To withdraw only the values a developer would use the ruby method ".values". For example if we only wanted the values from a hash called "directions" we would type in the following to get the values in that hash: p directions.values. This would return only the values and not the keys.
 
```

Comment:
Nice job remembering the `.values` method! However, the question is asking how can you extract a value, not all values. Minor thing, but always good to keep the attention to detail high.